packages/gatsby-source-filesystem/src/create-remote-file-node.js

```diff
@@ -178,8 +181,31 @@ const requestRemoteNode = (url, headers, tmpFilename, httpOpts, attempt = 1) =>
 
     responseStream.on(`response`, response => {
       resetTimeout()
+      contentLength =
+        response.headers && Number(response.headers[`content-length`])
 
       fsWriteStream.on(`finish`, () => {
+        // We have an incomplete download
+        if (contentLength && contentLength !== fsWriteStream.bytesWritten) {
+          fs.removeSync(tmpFilename)
+
+          if (attempt < INCOMPLETE_RETRY_LIMIT) {
```

Comment:
I would only use 1 limit. Could you explain why 2 seperate once are beneficial?
```suggestion
          if (attempt < STALL_RETRY_LIMIT) {
```

Reply:
@wardpeet Because one is for stalling and the other is for incomplete downloads 😅 I figured I would keep them separate. What do you think?

Reply:
Also, the current option is called `GATSBY_STALL_RETRY_LIMIT` and mentions stall. I don't want to break current users but using that limit for incomplete checks would be misleading.